numpyro/distributions/continuous.py

```diff
@@ -651,6 +651,156 @@ def variance(self):
+    where :math:`C` is the capacitance matrix :math:`I + W.T @ inv(D) @ W`, to compute
+    the log determinant.
+    """
+    sign_det_D, log_det_D = np.linalg.slogdet(D)
+    return 2 * np.sum(np.log(np.diagonal(capacitance_tril, axis1=-2, axis2=-1)), axis=-1) + sign_det_D * log_det_D
+
+
+def _batch_lowrank_mahalanobis(W, D, x, capacitance_tril):
+    r"""
+    Uses "Woodbury matrix identity"::
+        inv(W @ W.T + D) = inv(D) - inv(D) @ W @ inv(C) @ W.T @ inv(D),
+    where :math:`C` is the capacitance matrix :math:`I + W.T @ inv(D) @ W`, to compute the squared
+    Mahalanobis distance :math:`x.T @ inv(W @ W.T + D) @ x`.
+    """
+    Wt_Dinv = np.transpose(W, axes=[-1, -2]) / np.expand_dims(D, -2)
+    Wt_Dinv_x = _batch_mv(Wt_Dinv, x)
+    mahalanobis_term1 = np.sum(np.square(x) / D, axis=-1)
+    mahalanobis_term2 = _batch_mahalanobis(capacitance_tril, Wt_Dinv_x)
+    return mahalanobis_term1 - mahalanobis_term2
+
+
+@copy_docs_from(Distribution)
+class LowRankMultivariateNormal(Distribution):
+    arg_constraints = {"loc": constraints.real, "cov_factor": constraints.real, "cov_diag": constraints.positive}
+    support = constraints.real
+
+    def __init__(self, loc, cov_factor, cov_diag, validate_args=None):
+        if loc.ndim < 1:
+            raise ValueError("`loc` must be at least one-dimensional.")
+        event_shape = loc.shape[-1:]
+        if cov_factor.ndim < 2:
+            raise ValueError("`cov_factor` must be at least two-dimensional, "
+                             "with optional leading batch dimensions")
+        if cov_factor.shape[-2:-1] != event_shape:
+            raise ValueError("`cov_factor` must be a batch of matrices with shape {} x m"
+                             .format(event_shape[0]))
+        if cov_diag.shape[-1:] != event_shape:
+            raise ValueError("`cov_diag` must be a batch of vectors with shape {}".format(event_shape))
+
+        loc_ = np.expand_dims(loc, -1)
+        cov_diag_ = np.expand_dims(cov_diag, -1)
+        try:
+            loc_, self.cov_factor, cov_diag_ = np.broadcast_arrays(loc_, cov_factor, cov_diag_)
+        except RuntimeError:
+            raise ValueError("Incompatible batch shapes: loc {}, cov_factor {}, cov_diag {}"
+                             .format(loc.shape, cov_factor.shape, cov_diag.shape))
+        self.loc = loc_[..., 0]
+        self.cov_diag = cov_diag_[..., 0]
+        batch_shape = self.loc.shape[:-1]
+        self._unbroadcasted_cov_factor = cov_factor
+        self._unbroadcasted_cov_diag = cov_diag
+        self._capacitance_tril = _batch_capacitance_tril(cov_factor, cov_diag)
+        super(LowRankMultivariateNormal, self).__init__(batch_shape, event_shape, validate_args=validate_args)
+
+    @property
+    def mean(self):
+        return self.loc
+
+    @lazy_property
+    def variance(self):
+        # no expand needed here ??
+        return np.sum(np.square(self._unbroadcasted_cov_factor), axis=-1) + self._unbroadcasted_cov_diag
```

Comment:
I think we need to broadcast the result to `self.batch_shape + self.event_shape`, use `np.broadcast_to(..., self.batch_shape + self.event_shape)`

Reply:
@fehiepsi curious question for my own understanding: why do we broadcast `self.batch_shape + self.event_shape` here but only broadcast `self.batch_shape` for entropy calculation ?

Reply:
[entropy](https://en.wikipedia.org/wiki/Differential_entropy) is computed from the probability density function `log_prob` (which have shape `batch_shape`), hence it will have shape `batch_shape`. `variance` (similar to `mean`) is computed from `sample`s, hence it will have shape `batch_shape + event_shape`. I think this is true for all distributions.